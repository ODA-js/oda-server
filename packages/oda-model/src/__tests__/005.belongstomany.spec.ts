import 'jest';
import {
  BelongsToMany,
  belongsToManyDefaultMetaInfo,
  belongsToManyDefaultInput,
} from '../belongstomany';
import { merge } from 'lodash';

describe('belongsToMany', () => {
  it('default', () => {
    const res = new BelongsToMany({ belongsToMany: 'A#id', using: 'AB#' });
    expect(res.metadata.persistence).toMatchObject(
      belongsToManyDefaultMetaInfo.persistence,
    );
    expect(res.toObject()).toMatchObject(
      merge(
        { belongsToMany: 'id@A#id', using: 'id@AB#id' },
        belongsToManyDefaultInput,
      ),
    );
    expect(res.modelType).toBe('BelongsToMany');
    expect(res.verb).toBe('BelongsToMany');
    expect(res.toObject()).toMatchSnapshot('toObject');
  });
});
