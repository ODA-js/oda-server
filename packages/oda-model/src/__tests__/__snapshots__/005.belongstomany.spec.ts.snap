// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`belongsToMany default: toObject 1`] = `
Object {
  "belongsToMany": "id@A#id",
  "entity": undefined,
  "field": undefined,
  "fields": Array [],
  "metadata": Object {
    "name": Object {
      "full": "BelongsToMany",
      "normal": "",
      "short": "",
    },
    "persistence": Object {
      "embedded": false,
      "single": false,
      "stored": false,
    },
    "verb": "BelongsToMany",
  },
  "name": "BelongsToMany",
  "opposite": undefined,
  "using": "id@AB#id",
}
`;
